spec/functions/getpwnam_spec.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'stdlib::getpwnam' do
+  it { is_expected.not_to be_nil }
+  it { is_expected.to run.with_params.and_raise_error(ArgumentError, %r{expects 1 argument, got none}i) }
+
+  it {
+    passwd_entry = Etc::Passwd.new(
+       'steve',
+       'x',
+       1000,
+       1001,
+       'Steve',
+       '/home/steve',
+       '/bin/fish',
+     )
+    allow(Etc).to receive(:getpwnam).with(anything, anything).and_call_original
+    allow(Etc).to receive(:getpwnam).with('steve').and_return(passwd_entry)
+    is_expected.to run.with_params('steve').and_return(
+      {
+        'name' => 'steve',
+        'passwd' => 'x',
+        'uid' => 1000,
+        'gid' => 1001,
+        'gecos' => 'Steve',
+        'dir' => '/home/steve',
+        'shell' => '/bin/fish'
+      },
+    )
```

Comment:
Not sure if mocking is the best strategy.  I would only have a unit test (and skip it on windows) that get info for a user we are sure will exist (either we get the name of the current user or we look for `root`) and check that the returned hash include `{name: 'root'}`